velox/experimental/cudf/exec/ExpressionEvaluator.cpp

```diff
@@ -136,6 +139,68 @@ variant getVariant(const VectorPtr& vector, size_t atIndex = 0) {
   }
 }
 
+template <typename T>
+std::unique_ptr<cudf::scalar>
+makeScalarFromValue(const TypePtr& type, T value, bool isNull) {
```

Comment:
This appears to duplicate functionality of `makeScalarAndLiteral` above but this is neater. Can you please consolidate. You can use this function to create the scalar and make a convenience function to create ast literal.